src/main/java/org/opensearch/knn/index/query/parser/MethodParametersParser.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.query.parser;
+
+import com.google.common.collect.ImmutableSet;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.knn.index.query.model.AlgoQueryParameters;
+import org.opensearch.knn.index.query.model.HNSWAlgoQueryParameters;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER;
+import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER_EF_SEARCH;
+import static org.opensearch.knn.index.IndexUtil.isClusterOnOrAfterMinRequiredVersion;
+import static org.opensearch.knn.index.query.KNNQueryBuilder.EF_SEARCH_FIELD;
+import static org.opensearch.knn.index.query.KNNQueryBuilder.NAME;
+import static org.opensearch.knn.index.query.parser.KNNXParserUtil.parseJsonObject;
+
+@Getter
+public class MethodParametersParser {
+
+    private static final Set<String> VALID_METHOD_PARAMETERS = ImmutableSet.of(METHOD_PARAMETER_EF_SEARCH);
+
+    @SneakyThrows(IOException.class)
+    public static AlgoQueryParameters streamInMethodParameters(StreamInput in) {
```

Comment:
Here is an interesting choice of naming, we are calling function as streamInMethodParameters and rather then returning method parameters it is returning a class named as AlgoQueryParameters. We should be consistent in terms of what parameters these are. Are they algo related parameters or they are method related parameters. If you think algo and method is same thing, so use consistent naming.

Reply:
So the thought process was to make sure nothing other than algoquery parameters are there in this. From method parameters discussion things like overscorer can potential fit there but they shouldn't be clubbed and be added under algo params. The naming was chosen keeping in mind that. If something else get added in method params we should think about where to put it rather than using algo params as default. I can take a look at the structure again to see if I can make it more obvious